internal/pkg/pluginengine/change_helper.go

```diff
@@ -84,17 +86,16 @@ func changesForApply(smgr statemanager.Manager, cfg *configloader.Config) ([]*Ch
 				}
 
 				if resource == nil {
-					// tool exists in state, but resource doesn't exist, Create
-					changes = append(changes, generateCreateAction(&tool))
-					log.Infof("Change added: %s -> %s", tool.Name, statemanager.ActionCreate)
+					// tool exists in the state, but resource doesn't exist, Create
+					description := fmt.Sprintf("Tool < %s > state found but it seems the tool isn't created, will be created.", tool.Name)
+					changes = append(changes, generateCreateAction(&tool, description))
 				} else if drifted(resource, state.Resource) {
-					log.Debugf("Tool %s %s resource drifted from state.", tool.Name, tool.Plugin.Kind)
 					// resource drifted from state, Update
-					changes = append(changes, generateUpdateAction(&tool))
-					log.Infof("Change added: %s -> %s", tool.Name, statemanager.ActionUpdate)
+					description := fmt.Sprintf("Tool < %s > drifted from the state, will be updated.", tool.Name)
```

Comment:
Perhaps the 'state' information is invisible to the user?

Reply:
Very good point. I don't have a better description here; the main purpose is to let the user know why the "Upgrade" change is required. If we put only "Tool xxx will be updated", the user might think: why?

Is there a better way to put it? 
@lfbdev @daniel-hutao

Reply:
I  agree leave the state here for now.